classes/author/Repository.php

```diff
@@ -159,11 +159,11 @@ public function add(Author $author): int
             $author->setData('seq', $nextSeq);
         }
 
+        Hook::call('Author::add', [$author]);
```

Comment:
There's a `Hook::call('Author::delete::before', [$author]);`, so instead of moving this block (I consider it a breaking change), I think it's better to follow the pattern and add an `Author::add::before`.

Given the other "repositories" are following the same pattern, it would be better to update them as well (the `Author::edit` is also missing a `before` hook).

Reply:
Ok, I understand. 
Even if introducing breaking change is not really a problem :)
 I have developed some plugins for ojs 3.3 and I have to update them to follow changes between 3.3 and 3.4

Reply:
I agree, the codebase had way too many breaking changes. Hopefully it will decrease, it's really tough to understand what you've got to change without a well-written changelog.